source/analysis/AM1DNormalizationAB.h

```diff
@@ -87,7 +87,10 @@ class AM1DNormalizationAB : public AMStandardAnalysisBlock
 	/// When the independent values along an axis is not simply the axis index, this returns the independent value along an axis (specified by axis number and index)
 	virtual AMNumber axisValue(int axisNumber, int index) const;
 	/// Performance optimization of axisValue():  instead of a single value, copies a block of values from \c startIndex to \c endIndex in \c outputValues.  The provided pointer must contain enough space for all the requested values.
-    virtual bool axisValues(int axisNumber, int startIndex, int endIndex, double *outputValues) const;
+	virtual bool axisValues(int axisNumber, int startIndex, int endIndex, double *outputValues) const;
+
+	/// Returns the cached range of the data contained within the data source.  This is always valid because it is always recomputed when the data is recomputed.
+	virtual AMRange dataRange() const { return cachedDataRange_; }
```

Comment:
Well, it kind does that already.  Unfortunately, AMStandardAnalysisBlock is subclassed by more AB than just what we could call an AMStandardCachedAB where it would be able to take advantage of that kind of paradigm.  Now that someone else has some database upgrade experience, maybe the time do some refactoring is coming closer than maybe originally thought? ;)